lib/screens/first_screen.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_localization/navigation/app_routes.dart';
+import '../data/album.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class FirstScreen extends StatefulWidget {
+  const FirstScreen({Key? key}) : super(key: key);
+
+  @override
+  State<FirstScreen> createState() => _FirstScreenState();
+}
+
+class _FirstScreenState extends State<FirstScreen> {
+  Album? albumItem;
+  String titleText = "";
+  String bodyText = "";
+
+  @override
+  void initState() {
+    super.initState();
+    Future.delayed(Duration.zero, () {
+      setState(() {
+        titleText = AppLocalizations.of(context)!.nothingToShow;
+        bodyText = AppLocalizations.of(context)!.nothingToShow;
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(AppLocalizations.of(context)!.firstScreen),
+      ),
+      body: Center(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            ElevatedButton(
+              onPressed: () {
+                _navigateAndGetData(context);
+              },
+              child: Text(AppLocalizations.of(context)!.goToSecondScreen),
+            ),
+            Card(
+              child: ListTile(
+                title: Text(titleText),
+                subtitle: Text(bodyText),
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _navigateAndGetData(BuildContext context) async {
+    final result =
+        await Navigator.of(context).pushNamed(AppRoutes.secondScreen);
+    albumItem = result as Album;
+    updateTextData(albumItem);
```

Comment:
Here you can use `updateTextData(result as Album);` which is non-null and avoid null check in updateTextData